package/cloudshell/orch/training/services/sandbox_lifecycle.py

```diff
@@ -86,21 +87,19 @@ def end_student_reservation(self, user: str) -> None:
         if user_reservation_status == 'Completed':
             return
 
-        # get list of all apps that were deployed in instructor sandbox
-        # todo - when student mode need to use "createdInReservation" prop to get instructor deployed apps
-        instructor_resources = self._api.GetReservationDetails(self._sandbox.id).ReservationDescription.Resources
-        instructor_deployed_apps_names = [resource.Name for resource in instructor_resources if resource.VmDetails]
+        user_resources = self._api.GetReservationDetails(user_reservation_id).ReservationDescription.Resources
+        user_deployed_apps_names = [resource.Name for resource in user_resources if resource.VmDetails]
+        apps_names_shared_with_student = [app_name for app_name in user_deployed_apps_names if
+                                    self._api.GetResourceDetails(app_name).CreatedInReservation != user_reservation_id]
```

Comment:
Why do we need to do api.GetResourceDetails(app_name) for each app?
I think the CreatedInReservation prop should be returned in GetReservationDetails...